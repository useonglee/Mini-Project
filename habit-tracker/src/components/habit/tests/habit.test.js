import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";

import Habit from "../../habit/habit";

describe("habit component", () => {
  const habit = { name: "Habit", count: 4 };
  let HabitComponent;
  let handleIncrement;
  let handleDecrement;
  let handleDelete;

  beforeAll(() => {
    handleIncrement = jest.fn();
    handleDecrement = jest.fn();
    handleDelete = jest.fn();

    HabitComponent = (
      <Habit
        habit={habit}
        onIncrement={handleIncrement}
        onDecrement={handleDecrement}
        onDelete={handleDelete}
      />
    );
  });

  it("render habit component", () => {
    // snapshot testing
    const component = renderer.create(HabitComponent);
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe("Button Click", () => {
    it("calld handleIncrement when button is clicked", () => {
      render(HabitComponent);

      const button = screen.getByTitle("increase");
      userEvent.click(button);

      expect(handleIncrement).toHaveBeenCalledTimes(1);
      expect(handleIncrement).toHaveBeenCalledWith(habit);
    });
  });
});
