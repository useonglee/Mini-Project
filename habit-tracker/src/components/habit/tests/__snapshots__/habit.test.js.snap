// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`habit component render habit component 1`] = `
<li
  className="habit"
>
  <span
    className="habit-name"
    data-testid="habit-name"
  >
    Habit
  </span>
  <span
    className="habit-count"
    data-testid="habit-count"
  >
    4
  </span>
  <button
    className="habit-button habit-increase"
    onClick={[Function]}
    title="increase"
  >
    <i
      className="fas fa-plus-square"
    />
  </button>
  <button
    className="habit-button habit-decrease"
    onClick={[Function]}
    title="decrease"
  >
    <i
      className="fas fa-minus-square"
    />
  </button>
  <button
    className="habit-button habit-delete"
    onClick={[Function]}
    title="delete"
  >
    <i
      className="fas fa-trash"
    />
  </button>
</li>
`;
